src/app/core/layers/providers/kmlprovider.js

```diff
@@ -11,9 +11,7 @@ inherit(KMLDataProvider, DataProvider);
 const proto = KMLDataProvider.prototype;
 
 proto.getData = function() {
-  const d = $.Deferred();
-  return d.promise();
+  return new Promise((resolve, reject) => {})
```

Comment:
double check, how can such a promise object be fulfilled?

```js
return new Promise((resolve, reject) => {})
```

Reply:
@Raruto i think that KMLDataProvider is never used. We need to double check and remove it

Reply:
ok, i didn't check in too much detail, but still from some other substitutions you've done i guess your initial intent was to write something like this instead:

```js
Promise.resolve()
```